src/esolver/lis_esolver_ai.c

```diff
@@ -324,17 +324,17 @@ LIS_INT lis_eai(LIS_ESOLVER esolver)
       lis_printf(comm,"computing refined eigenpairs using inner eigensolver:\n\n");
     }
 
-  lis_esolver_create(&esolver2);
-  esolver2->options[LIS_EOPTIONS_ESOLVER] = niesolver;
-  esolver2->options[LIS_EOPTIONS_SUBSPACE] = 1;
-  esolver2->options[LIS_EOPTIONS_MAXITER] = emaxiter;
-  esolver2->options[LIS_EOPTIONS_OUTPUT] = esolver->options[LIS_EOPTIONS_OUTPUT];
-  esolver2->params[LIS_EPARAMS_RESID - LIS_EOPTIONS_LEN] = tol;
-
   /* compute refined (real) eigenpairs */
```

Comment:
The `esolver2` is allocated once but used multiple times within the loop.
The `esolver2->evector` is reallocated later in the call `lis_esolve(A, esolver->evector[i], &evalue, esolver2);` via `lis_vector_duplicate(A, &esolver->evector[j]);` in the solving logic.
But they are never freed before duplication.
This causes memory leak.

Two ways to address the issue.
1) destroy and recreate the solver as the modification does.
2) provide a mechanism to partially clean `esolver2`, something between `lis_esolver_work_destroy` and `lis_esolver_destroy`, that also cleans `evector`.